src/CareTogether.Api/Startup.cs

```diff
@@ -320,6 +320,25 @@ out var researchOrganizationId
                                     return;
                                 }
 
+                                if (

```

Comment:
This whole basic authentication implementation is vulnerable to timing attacks. Can you add a random delay (`await Task.Delay(...)`) from between 50-100ms at the beginning of the `OnValidateCredentials` lambda (line 287)? See also https://learn.microsoft.com/en-us/dotnet/api/System.Random?view=net-9.0